ext/afform/core/Civi/Api4/Action/Afform/AbstractProcessor.php

```diff
@@ -241,6 +241,14 @@ public function loadEntity(array $entity, array $values, string $mode = 'update'
       if (!empty($result[$key])) {
         $data = ['fields' => $result[$key]];
         foreach ($entity['joins'] ?? [] as $joinEntity => $join) {
+          // Based on Civi\Api4\Action\Afform\Get::getCustomGroupBlocks(), we can use this check for "is_multiple"
+          // This is also a way to check if the join entity is a custom block
+          $isMultiRecord = str_contains($joinEntity, "Custom_");
+          $joinAllowedAction = self::getJoinAllowedAction($entity, $joinEntity);
+          // Do not load joins when block action is create
+          if ($isMultiRecord === TRUE && $joinAllowedAction["create"] === TRUE) {
```

Comment:
If multirecord and Create is checked, do not load joins

Reply:
The idea with my PR https://github.com/civicrm/civicrm-core/pull/31480 is that `$isMultiRecord` is now moot, we can just go off the allowed action and treat all joins equally.